### PR TITLE
Jira/crud/aman

### PR DESCRIPTION
**Pull request by aman5275555fff**
crud operation Application, Module, Workflow, WorkflowStage, WorkflowActivitiy, Organization table

### CODE REVIEW

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/ModuleRepository.cs [hunk 1]
```diff
@@ -5,17 +5,117 @@
...
             _connectionFactory = connectionFactory;
         }
+
+        public List<Dictionary<int, string>> GetValues(int moduleId)
+        {
+            using IDbConnection con = _connectionFactory.GetConnection;
+            var query = @"select r.Id as Keys , upd.Value as [Values] From Record r 
+                            Inner Join UiPageData upd on r.Id = upd.RecordId
+                        where r.ModuleId = @moduleId and upd.UiPageMetadataId = 3014 and r.IsDeleted = 0 and upd.IsDeleted = 0";
+            var df = con.Query<Dictionary<int, string>>(query, new { moduleId }).ToList();
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/ModuleRepository.cs:32]: what is this method and why it is having hard coded UiPageMetadataId? and why does it return Dectionary.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/ModuleRepository.cs [hunk 2]
```diff
@@ -5,17 +5,117 @@
...
+        /// <returns></returns>
+        public bool Delete(int id)
+        {
+            using IDbConnection db = _connectionFactory.GetConnection;
+
+            db.Execute(@"update [Module] Set 
+                                    IsDeleted = 1
+                                    Where Id = @Id", new { Id = id });
+            return true;
+        }
```
Comment (smartdusttechnologies) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/ModuleRepository.cs:118]: why are we not using generic repository where we only need one table query.

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ActivityService.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        public ActivityModel GetById(int id)
+        {
+            return _genericRepository.Get(id);
+        }
+        #endregion
+
+        #region Private Methods
+
+        #endregion
+    }
```
[the remark this answers is not in the document]
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ActivityService.cs:76]: what are you doing aman you have not resolved my comment and then why resolved this comment by yourself 
pehle resolve karo comment then inform me tumhe resulve karne ki permission ni hia

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ApplicationServices.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        public ApplicationModel GetById(int id)
+        {
+            return _genericRepository.Get(id);
+        }
+        #endregion
+
+        #region Private Methods
+
+        #endregion
+    }
```
[the remark this answers is not in the document]
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ApplicationServices.cs:72]: why not using DTO here

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/WorkflowActivityService.cs [hunk 5]
```diff
@@ -19,42 +20,91 @@ public class WorkflowActivityService : IWorkflowActivityService
...
-            //var activityList = _workflowActivityRepository.GetByWorkflowStageId(recordModel.WorkflowStageId);
-            //var pageDataList = _pageDataGenericRepository.Get("RecordId", recordModel.Id);
-            //foreach (var activity in activityList)
-            //{
-            //    if (activity.ActivityId == (int)ActivityType.EmailServices)
-            //    {
-            //       SendEmail(pageDataList,activity.ActivityId,recordModel.WorkflowStageId);
-            //    }
-            //}
-            return true;
```
[the remark this answers is not in the document]
Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/WorkflowActivityService.cs:52]: aman what are you doing why deleting existing method revert this changes

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/WorkflowStageDTO.cs [hunk 6]
```diff
@@ -1,11 +1,42 @@
...
+{   /// <summary>
+    /// It Contains The Properties for WorkflowStage
+    /// </summary>
     public class WorkflowStageDTO
-    {
+    {    /// <summary>
+         /// It Contains The Id of WorkflowStage
+         /// </summary>
         public int Id { get; set; }
+        /// <summary>
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/WorkflowStageDTO.cs:12]: Why using id Use Entity

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/WorkflowDTO.cs [hunk 7]
```diff
@@ -1,9 +1,27 @@
...
+{    /// <summary>
+     /// It Conatains The Properties for Workflow
+     /// </summary>
     public class WorkflowDTO
-    {
+    {    /// <summary>
+         /// It Contains The Id of Workflow
+         /// </summary>
         public int Id { get; set; }
+        [Required(ErrorMessage = "Please Enter Name")]
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/WorkflowDTO.cs:12]: Why using id Use Entity

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/WorkflowActivityDTO.cs [hunk 8]
```diff
@@ -1,12 +1,36 @@
...
+{    /// <summary>
+     /// It Conatains The Properties for WorkflowActivity
+     /// </summary>
     public class WorkflowActivityDTO
-    {
+    {     /// <summary>
+          /// It Contains The Id of WorkflowActivity
+          /// </summary>
         public int Id { get; set; }
+        /// <summary>
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/WorkflowActivityDTO.cs:12]: Why using id Use Entity

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/OrganizationDTO.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+namespace TestingAndCalibrationLabs.Web.UI.Models
+{   /// <summary>
+    /// It Conatains The Properties for Organization
+    /// </summary>
+    public class OrganizationDTO
+    {    /// <summary>
+         /// It Contains The Id of Organization
+         /// </summary>
+        public int Id { get; set; }
+        /// <summary>
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/OrganizationDTO.cs:12]: Why using id Use Entity

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/ModuleDTO.cs [hunk 10]
```diff
@@ -1,9 +1,27 @@
...
+{    /// <summary>
+     /// It Conatains The Properties for Module
+     /// </summary>
     public class ModuleDTO
-    {
+    {   /// <summary>
+        /// It Contains The Id of Module
+        /// </summary>
         public int Id { get; set; }
+        /// <summary>
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/ModuleDTO.cs:12]: Why using id Use Entity

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/ApplicationDTO.cs [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+{
+    /// <summary>
+    /// It Conatains The Properties for Application
+    /// </summary>
+    public class ApplicationDTO
+    {
+        /// <summary>
+        /// It Contains The Id of Application
+        /// </summary>
+        public int Id { get; set; }
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/ApplicationDTO.cs:13]: Why using id Use Entity

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/ActivityDTO.cs [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+{
+    /// <summary>
+    /// It Conatains The Properties for  Activity
+    /// </summary>
+    public class ActivityDTO
+    {
+        /// <summary>
+        /// It Contains The Id of  Activity
+        /// </summary>
+        public int Id { get; set; }
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Models/Common/ActivityDTO.cs:13]: Why using id Use Entity

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs [hunk 13]
```diff
@@ -123,6 +127,15 @@ public void ConfigureServices(IServiceCollection services)
...
             services.AddScoped<IGenericRepository<UiControlTypeModel>, GenericRepository<UiControlTypeModel>>();
             services.AddScoped<IGenericRepository<UiPageMetadataModel>, GenericRepository<UiPageMetadataModel>>();
+            services.AddScoped<IGenericRepository<ApplicationModel>, GenericRepository<ApplicationModel>>();
+            services.AddScoped<IGenericRepository<ActivityModel>, GenericRepository<ActivityModel>>();
+            services.AddScoped<IGenericRepository<Organization>, GenericRepository<Organization>>();
+            services.AddScoped<IGenericRepository<WorkflowActivityModel>, GenericRepository<WorkflowActivityModel>>();
+            services.AddScoped<IGenericRepository<WorkflowStageModel>, GenericRepository<WorkflowStageModel>>();
+
+            ////
+            ///
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:137]: remove commented code from here

Comment (rajguptaH) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Startup.cs:137]: ??

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowStageRepository.cs [hunk 14]
```diff
@@ -8,11 +8,38 @@ namespace TestingAndCalibrationLabs.Business.Data.Repository.Interfaces
...
+        /// </summary>
+        /// <param name="workflowStageModel"></param>
+        /// <returns></returns>
+        int Create(WorkflowStageModel workflowStageModel);
+        /// <summary>
+        /// Update Record In WorkflowStage
+        /// </summary>
+        /// <param name="workflowStageModel"></param>
+        /// <returns></returns>
+        int Update(WorkflowStageModel workflowStageModel);
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowStageRepository.cs:43]: delete is generic

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/WorkflowStageController.cs [hunk 15]
```diff
@@ -4,34 +4,157 @@
...
+            var workflowList = _workflowService.Get();
+            var pages = _mapper.Map<List<UiPageTypeModel>, List<UiPageTypeDTO>>(pageList);
+            var workflowLists = _mapper.Map<List<WorkflowModel>, List<WorkflowDTO>>(workflowList);         
+            //var dropdownssss =    categories.Select(x => new SelectListItem { Text = x.Name, Value = x.Id.ToString() }).ToList();
+            ViewBag.UiPageTypes = pages;
+            ViewBag.Workflow = workflowLists;        
+            return base.View(new Models.WorkflowStageDTO { Id = id });
+        }
+        /// <summary>
+        /// To Create Record In WorkflowActivity
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/WorkflowStageController.cs]: not commented by m

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Tests/bin/Debug/netcoreapp3.1/appsettings.json [hunk 16]
```diff
@@ -9,7 +9,7 @@
   },
```
Comment (rajguptaH) [inline TestingAndCalibrationLabs/TestingAndCalibrationLabs.Tests/bin/Debug/netcoreapp3.1/appsettings.json:9]: why this file is commited

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/WorkflowActivityRepository.cs [hunk 17]
```diff
@@ -32,5 +32,96 @@ public List<WorkflowActivityModel> GetByWorkflowStageId(int stageId)
...
+        /// <returns></returns>
+        //public bool Delete(int id)
+        //{
+        //    using IDbConnection db = _connectionFactory.GetConnection;
+
+        //    db.Execute(@"update [WorkflowActivity] Set 
+        //                            IsDeleted = 1
+        //                            Where Id = @Id", new { Id = id });
+        //    return true;
+        //}
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/WorkflowActivityRepository.cs]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/WorkflowRepository.cs [hunk 18]
```diff
@@ -28,6 +28,90 @@ public WorkflowModel GetByModuleId(int moduleId)
...
+        /// <returns></returns>
+        //public bool Delete(int id)
+        //{
+        //    using IDbConnection db = _connectionFactory.GetConnection;
+
+        //    db.Execute(@"update [Workflow] Set 
+        //                            IsDeleted = 1
+        //                            Where Id = @Id", new { Id = id });
+        //    return true;
+        //}
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/WorkflowRepository.cs]: made delete generic

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/WorkflowStageRepository.cs [hunk 19]
```diff
@@ -9,25 +9,113 @@
...
+        /// <returns></returns>
+        //public bool Delete(int id)
+        //{
+        //    using IDbConnection db = _connectionFactory.GetConnection;
+
+        //    db.Execute(@"update [WorkflowStage] Set 
+        //                            IsDeleted = 1
+        //                            Where Id = @Id", new { Id = id });
+        //    return true;
+        //}
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/WorkflowStageRepository.cs]: made delete generic

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IModuleRepository.cs [hunk 20]
```diff
@@ -1,6 +1,45 @@
...
+        /// </summary>
+        /// <param name="moduleModel"></param>
+        /// <returns></returns>
+        int Update(ModuleModel moduleModel);
+        /// <summary>
+        /// Delete Record From Module By Id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IModuleRepository.cs]: delete is made generic

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowActivityRepository.cs [hunk 21]
```diff
@@ -13,6 +13,36 @@ public interface IWorkflowActivityRepository
...
+        /// </summary>
+        /// <param name="workflowActivityModel"></param>
+        /// <returns></returns>
+        int Update(WorkflowActivityModel workflowActivityModel);
+        /// <summary>
+        /// Delete Record From WorkflowActivity By Id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+      //  bool Delete(int id);
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowActivityRepository.cs]: delete is generic

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowRepository.cs [hunk 22]
```diff
@@ -12,6 +13,35 @@ public interface IWorkflowRepository
...
+        /// </summary>
+        /// <param name="workflowModel"></param>
+        /// <returns></returns>
+        int Update(WorkflowModel workflowModel);
+        /// <summary>
+        /// Delete Record From Workflow By Id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        //bool Delete(int id);
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowRepository.cs]: delete is generic

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowStageRepository.cs [hunk 23]
```diff
@@ -8,11 +8,42 @@ namespace TestingAndCalibrationLabs.Business.Data.Repository.Interfaces
...
+        /// <summary>
+        /// Delete Record From WorkflowStage By Id
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+      //  bool Delete(int id);
+
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowStageRepository.cs]: delete is generic

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/ActivityModel.cs [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+    /// It Conatains The Properties for Activity
+    /// </summary>
+    [DbTable("Activity")]
+    public class ActivityModel : Entity
+    {
+        /// <summary>
+        /// It Contains Name Of Activity
+        /// </summary>
+        public string Name { get; set; }
+
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/ActivityModel.cs]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/ApplicationModel.cs [hunk 25]
```diff
@@ -0,0 +1,22 @@
...
+    /// It Conatains The Properties for Application
+    /// </summary>
+    [DbTable("Application")]
+    public class ApplicationModel : Entity
+    {
+        /// <summary>
+        /// It Contains Name Of Application
+        /// </summary>
+        public string Name { get; set; }
+
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Core/Model/Common/ApplicationModel.cs]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/ModuleRepository.cs [hunk 26]
```diff
@@ -5,17 +5,84 @@
...
+        /// Getting Record By Id Module
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public ModuleModel GetById(int id)
+        {
+            using IDbConnection db = _connectionFactory.GetConnection;
+            var moduleById = db.Query<ModuleModel>(@"Select m.Id,
+                                                       m.ApplicationId,     
+													 a.[Name] as ApplicationName,
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/BackOffice/ModuleRepository.cs:63]: in my system showing properly

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowActivityRepository.cs [hunk 27]
```diff
@@ -8,11 +8,33 @@ namespace TestingAndCalibrationLabs.Business.Data.Repository.Interfaces
...
     public interface IWorkflowActivityRepository
     {
-     /// <summary>
-     /// Get All Records From 
-     /// </summary>
-     /// <returns></returns>
-     List<WorkflowActivityModel> GetByWorkflowStageId(int uiPageTypeId);
-       
+        /// <summary>
+        /// Get All Records From
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowActivityRepository.cs:12]: done

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowRepository.cs [hunk 28]
```diff
@@ -7,11 +8,33 @@ namespace TestingAndCalibrationLabs.Business.Data.Repository.Interfaces
...
     public interface IWorkflowRepository
     {
-     /// <summary>
-     /// Get All Records From 
-     /// </summary>
-     /// <returns></returns>
-     WorkflowModel GetByModuleId(int moduleId);
-       
+        /// <summary>
+        /// Get All Records From
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Data/Repository/Interfaces/BackOffice/IWorkflowRepository.cs:12]: done

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ActivityService.cs [hunk 29]
```diff
@@ -0,0 +1,77 @@
...
+        /// Insert Record In Activity
+        /// </summary>
+        /// <param name="activityModel"></param>
+        /// <returns></returns>
+        public RequestResult<int> Create(ActivityModel activityModel)
+        {
+            _genericRepository.Insert(activityModel);
+            return new RequestResult<int>(1);
+        }
+
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ActivityService.cs]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/ApplicationController.cs [hunk 30]
```diff
@@ -0,0 +1,143 @@
...
+        /// </summary>
+        /// <param name="applicationDTO"></param>
+        /// <returns></returns>
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult Create([Bind] ApplicationDTO applicationDTO)
+        {
+            if (ModelState.IsValid)
+            {
+
```
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/ApplicationController.cs]: removed

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/WorkflowStageController.cs (hunk 15 above)
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Web.UI/Controllers/WorkflowStageController.cs]: done

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Tests/bin/Debug/netcoreapp3.1/appsettings.json (hunk 16 above)
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Tests/bin/Debug/netcoreapp3.1/appsettings.json:9]: by mistake

File: TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ActivityService.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (aman5275555fff) [reply TestingAndCalibrationLabs/TestingAndCalibrationLabs.Business/Services/BackOffice/ActivityService.cs:67]: when i removed end region directective expected error occur

**Pull request merged**